### PR TITLE
INT-589 Call queue

### PR DESCRIPTION
**Pull request by nurutdinov**

### CODE REVIEW

File: talk-call-queue/talk-call-queue-api.yaml [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+        204:
+          description: The call queue has been deleted.
+          content: { }
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+
+  /webhooks/calls/enqueue/{id}:
+    post:
+      tags:
+        - Calls Webhooks
```
Comment (BullDog57Rus) [inline talk-call-queue/talk-call-queue-api.yaml:159]: check the tags from the first comment

File: talk-call-queue/talk-call-queue-api.yaml [hunk 2]
```diff
@@ -0,0 +1,739 @@
...
+      parameters:
+        - $ref: '#/components/parameters/ResourceId'
+      responses:
+        204:
+          description: The call queue has been deleted.
+          content: { }
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+
+  /webhooks/calls/enqueue/{id}:
```
[the remark this answers is not in the document]
Comment (BullDog57Rus) [reply talk-call-queue/talk-call-queue-api.yaml:156]: and below

File: talk-call-queue/talk-call-queue-api.yaml [hunk 3]
```diff
@@ -0,0 +1,739 @@
...
+      summary: Dequeue reason and details about the time spent in the queue.
+      operationId: action
+      parameters:
+        - $ref: '#/components/parameters/ResourceId'
+      requestBody:
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              properties:
+                QueueTime:
```
Comment (BullDog57Rus) [inline talk-call-queue/talk-call-queue-api.yaml:204]: required?

File: talk-call-queue/talk-call-queue-api.yaml [hunk 4]
```diff
@@ -0,0 +1,739 @@
...
+      tags:
+        - Calls Webhooks
+      x-access: private
+      summary: Receive status of the call.
+      operationId: operatorStatus
+      parameters:
+        - $ref: '#/components/parameters/ResourceId'
+        - $ref: '#/components/parameters/SubresourceId'
+      requestBody:
+        required: true
```
Comment (BullDog57Rus) [inline talk-call-queue/talk-call-queue-api.yaml:348]: check everywhere

File: talk-call-queue/talk-call-queue-api.yaml [hunk 5]
```diff
@@ -0,0 +1,739 @@
...
+      in: query
+      description: Query parameter for the "sort" criteria.
+      schema:
+        type: string
+
+  responses:
+
+    EmptyResponse:
+      description: Returns an empty response that is used for OK non-returning operations.
+      content: { }
```
Comment (BullDog57Rus) [inline talk-call-queue/talk-call-queue-api.yaml:413]:
```suggestion
```
and can be used in several requests above

File: talk-call-queue/talk-call-queue-api.yaml [hunk 6]
```diff
@@ -0,0 +1,739 @@
...
+        - callDistributionType
+        - wrapUpTime
+        - timezone
+        - twentyFourSeven
+        - businessHours
+        - maxSize
+        - maxWaitTime
+        - callbackAllowed
+        - promptVoice
+        - promptLanguage
```
Comment (BullDog57Rus) [inline talk-call-queue/talk-call-queue-api.yaml:561]: you can check, but I think `required` is not needed for properties with `default` value

**Review by BullDog57Rus — CHANGES_REQUESTED** [review]

File: talk-call-queue/talk-call-queue-api.yaml (hunk 6 above)
Comment (nurutdinov) [reply talk-call-queue/talk-call-queue-api.yaml:561]: Yes, agree, but Evgeniy Petukhov added it for UI API interface.

**Review by BullDog57Rus — APPROVED** [review]

**Pull request merged**